src/rtld_loader.c

```diff
@@ -48,7 +48,24 @@ char* la_objsearch(const char* name, uintptr_t* cookie, unsigned int flag) {
       log_info("\n  searching...\n");
       const char* result = NULL;
       if (result == NULL) {
-        result = dynamic_lookup(libname, replit_ld_library_path);
+        const char* search_path = replit_ld_library_path;
+        if (strstartswith("/nix/store/", name)) {
+          // Count slashes to find the 5th one (after /nix/store/hash-name/lib/)
```

Comment:
lol we can't use any libs.

Reply:
You mean the `strstartswith`?

Reply:
Just a comment we can't use a regex lib for example.